### ISSUE TITLE
building custom database fail at step 5 

### ISSUE THREAD

**Issue by dpchris** [DerrickWood/kraken#107]
Hi, 

I have already successfully build custom databases but for some reason, I got an error at step 5 with the following command :
kraken-build --threads 40 --build --db ~/kraken_customdb

Kraken build set to minimize disk writes.
Creating k-mer set (step 1 of 6)...
Found jellyfish v1.1.11
Hash size not specified, using '18503360000'
K-mer set created. [1h4m34.027s]
Skipping step 2, no database reduction requested.
Sorting k-mer set (step 3 of 6)...
K-mer set sorted. [2h1m17.745s]
Skipping step 4, GI number to seqID map now obsolete.
Creating seqID to taxID map (step 5 of 6)...
No preliminary seqid/taxid mapping files found, aborting.

Do you have any ideas of why ?

Thanks in advance.

Best regards.

**Comment by jenniferlu717** [comment]
How did you add your files to the library? Unfortunately you cannot just place the files directly into the library folder, you should use the add-to-library option. I believe if you just do this step, it will create the files needed to make the seqid2taxid file and you won't have to redo the previous build steps

**Comment by dpchris** [comment]
I added my files using add-to-library option :

for dir in fungi protozoa archaea viral bacteria; do
    for fna in `ls $dir/*.fna`; do
        kraken-build --add-to-library $fna --db ../kraken_customdb_28-11-17_modif_kurstaki
    done
done

and I downloaded the taxonomy with kraken-build --download-taxonomy

The headers of my fasta files looks like this :
">NC_021248.1|kraken:taxid|10288 Choristoneura biennis entomopoxvirus 'L', complete genome"

I successfully build my custom database with kraken v0.10.6 but I failed at step 5 with kraken v1.0 on the same database.

Maybe I used add-to-library with kraken 0.10.6 and try to build the database with kraken v.1.0, is that a problem ?

________________________________
De : jenniferlu717 [notifications@github.com]
Envoyé : mercredi 10 janvier 2018 19:05
À : DerrickWood/kraken
Cc : CHRISTIANY David; Author
Objet : Re: [DerrickWood/kraken] building custom database fail at step 5 (#107)


How did you add your files to the library? Unfortunately you cannot just place the files directly into the library folder, you should use the add-to-library option. I believe if you just do this step, it will create the files needed to make the seqid2taxid file and you won't have to redo the previous build steps

**Comment by jenniferlu717** [comment]
Yes I think so. Part of the fix to remove the dependence on gi numbers in kraken v1.0 requires that another file is generated when using the --add-to-library fix. If you redo this with the newest kraken version, it should fix the problem?

**Comment by jmonroynieto** [comment]
I had the same error message. I am leaving a note for any other users that stumble across this (as did I).

```
...
Skipping step 4, GI number to seqID map now obsolete.
Creating seqID to taxID map (step 5 of 6)...
No preliminary seqid/taxid mapping files found, aborting.
...
```

I am building a custom database with a custom phylogenetic taxonomy with only 26 nodes for a single species. My error was very simple: I was creating the preliminary map on the _taxonomy_ directory and it must be in the _library_ directory.

**Comment by Kami0King** [comment]
> I had the same error message. I am leaving a note for any other users that stumble across this (as did I).
> 
> ```
> ...
> Skipping step 4, GI number to seqID map now obsolete.
> Creating seqID to taxID map (step 5 of 6)...
> No preliminary seqid/taxid mapping files found, aborting.
> ...
> ```
> 
> I am building a custom database with a custom phylogenetic taxonomy with only 26 nodes for a single species. My error was very simple: I was creating the preliminary map on the _taxonomy_ directory and it must be in the _library_ directory.

I have the same issue, we added a IC(inner control sequences), I should build the IC seq database, after I added the IC.fasta in add-new-library step, I saw the prelim_map.txt under library/added, but its size was 0. Then I got stuck in 'No preliminary seqid/taxid mapping files found, aborting.'  step too.

**Comment by rachelmugge** [comment]
The above solution worked for me!
